x/stakeibc/keeper/validator_selection.go

```diff
@@ -0,0 +1,46 @@
+package keeper
+
+import (
+	"github.com/Stride-Labs/stride/x/stakeibc/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) GetValidatorAmtDifferences(ctx sdk.Context, hostZone types.HostZone) map[string]int64 {
+	/*
+		This function returns a map from Validator Address to how many extra tokens
+		need to be given to that validator (posit)
+	*/
+	validators := hostZone.GetValidators()
+	scaled_weights := make(map[string]int64)
+	target_weights := k.GetTargetValAmtsForHostZone(ctx, hostZone)
+	for _, validator := range validators {
+		scaled_weights[validator.Address] = target_weights[validator.Address] - validator.DelegationAmt
+	}
+	return scaled_weights
+}
+
+func (k Keeper) GetTargetValAmtsForHostZone(ctx sdk.Context, hostZone types.HostZone) map[string]int64 {
+	var out map[string]types.ValidatorWeights
+	k.paramstore.Get(ctx, types.KeyHostZoneValidatorWeights, &out)
+	valWeights := out[hostZone.ChainId]
+	totalWeight := int64(0)
+	totalDelegatedWeight := k.GetTotalValidatorWeight()
+	for _, weight := range valWeights.ValidatorWeights {
+		totalWeight += weight
+	}
+	var targetWeight map[string]int64
+	for valAddr, weight := range valWeights.ValidatorWeights {
+		targetWeight[valAddr] = int64(float64(weight*totalDelegatedWeight) / float64(totalWeight))
+	}
```

Comment:
I'm a little confused on "weight" vs "amt" (might just be me). Is "amt" the current weight and "weight" is the desired weight? If so, might not be a bad idea to rename accordingly

Reply:
good call, I agree we should at least comment to make this clear. i assume weight is a decimal \in[0,1] and amt=w*total